### PR TITLE
Move index data out of chain db. #1396

### PR DESCRIPTION
**Pull request by 5sWind**
Not a final commit version - not all tests covered. There are some comments wants to check.

### CODE REVIEW

File: blockchain/blockchain.go [hunk 1]
```diff
@@ -233,13 +233,14 @@ func PrecreatedDaoOption(dao *blockDAO) Option {
...
-// BoltDBDaoOption sets blockchain's dao with BoltDB from config.Chain.ChainDBPath
+// BoltDBDaoOption sets blockchain's dao with BoltDB from config.Chain.ChainDBPath and cfg.DB.IndexDBPath
 func BoltDBDaoOption() Option {
 	return func(bc *blockchain, cfg config.Config) error {
 		cfg.DB.DbPath = cfg.Chain.ChainDBPath // TODO: remove this after moving TrieDBPath from cfg.Chain to cfg.DB
 		_, gateway := cfg.Plugins[config.GatewayPlugin]
 		bc.dao = newBlockDAO(
-			db.NewBoltDB(cfg.DB),
+			db.NewBoltDB(cfg.DB, cfg.DB.DbPath),
+			db.NewBoltDB(cfg.DB, cfg.DB.IndexDBPath),
```
Comment (5sWind) [inline blockchain/blockchain.go]: Is it a doable way and does it make sense.

File: blockchain/blockdao.go [hunk 2]
```diff
@@ -120,6 +123,7 @@ func newBlockDAO(kvstore db.KVStore, writeIndex bool, compressBlock bool, maxCac
 	}
 	blockDAO.timerFactory = timerFactory
 	blockDAO.lifecycle.Add(kvstore)
+	blockDAO.lifecycle.Add(kvistore)
```
Comment (5sWind) [inline blockchain/blockdao.go]: same as above.

**Comment by 5sWind** [comment]
> Also I have an ongoing PR to improve the address/action indexing, which I will update by tmrw. To avoid duplicate work and/or merge conflict, could this PR delay a little bit to come after mine?

okay...

**Comment by dustinxie** [comment]
this will be done together with PR1461, both changes involve DB, better to combine into 1 single commit

**Pull request closed without merging**